components/org.wso2.carbon.identity.mgt.store.connector.jdbc/src/main/java/org/wso2/carbon/identity/mgt/store/connector/jdbc/queries/MySQLFamilySQLQueryFactory.java

```diff
@@ -285,16 +285,16 @@ public MySQLFamilySQLQueryFactory() {
 
     public String getQuerryForUserIdFromMultipleAttributes(List<Attribute> attributes, int offset, int length) {
         StringBuilder getUniqueUserQuerry = new StringBuilder();
-         getUniqueUserQuerry.append("SELECT UM_USER.USER_UNIQUE_ID FROM UM_USER WHERE UM_USER.ID IN");
+        getUniqueUserQuerry.append("SELECT UM_USER.USER_UNIQUE_ID FROM UM_USER WHERE UM_USER.ID IN");
         int count = 1;
-        for (Attribute attribute : attributes) {
+        while (count <= attributes.size()) {
             getUniqueUserQuerry
                     .append(" (SELECT UM_USER_ATTRIBUTES.USER_ID FROM UM_USER_ATTRIBUTES" +
-                            " WHERE ATTR_ID = (SELECT ID FROM UM_ATTRIBUTES WHERE ATTR_NAME = '")
-                    .append(attribute.getAttributeName())
-                    .append("' ) AND ATTR_VALUE = '")
-                    .append(attribute.getAttributeValue())
-                    .append("')");
+                            " WHERE ATTR_ID = (SELECT ID FROM UM_ATTRIBUTES WHERE ATTR_NAME = ")
+                    .append(":attri").append((count * 2) - 1)
+                    .append("; ) AND ATTR_VALUE = ")
+                    .append(":attri").append(count * 2)
+                    .append("; )");
```

Comment:
@GDRDABARERA I think this code is still vulnerable to SQL injection. You should use PreparedStatement instead to construct the SQL query. Is it not possible here? cc @Pushpalanka

Reply:
The issue is fixed now. The confusion might be resolved now